gemcalibration/src/common/CalibrationWeb.cc

```diff
@@ -0,0 +1,307 @@
+        *out << "                                <label> <input type=\"checkbox\" class=\"check\" name=\""<<  t_stream.str() << "\" id=\"" << t_stream.str() << "\">" << t_stream.str() << "</label>"<< std::endl;
+        *out << "                            </div>" << std::endl;
+        *out << "                        </div>"<< std::endl;//end column
+
+        for (auto dacScan_parameter: it.second.range) {
+            *out << "                        <div class=\"col-md-4\">" <<std::endl;
+            this->genericParamSelector_dacScan(  dacScan_parameter.first, dacScan_parameter.second, out);
+            *out << "                        </div>"<< std::endl;//end column
+        }
+        *out << "                    </div>" << std::endl; // end <div class="row">
+    } // end loop over DAC scan type
+    *out << "                </div>" << std::endl; // end drop-down scrollable
+    *out << "            </div>" << std::endl; // end drop-down
+    *out << "        </div>" << std::endl; // end container
+    *out << "    </form>"<< std::endl;
+    *out << "    <div align=\"center\">"<< std::endl;
+    *out << "        <button class=\"btn btn-lg btn-info\" onclick=\"default_dacscans()\">SELECT DEFAULT</button>" << std::endl;
+    *out << "        <button class=\"btn btn-lg btn-info\" onclick=\"select_dacscans()\">SELECT ALL</button>" << std::endl;
+    *out << "        <button class=\"btn btn-lg btn-warning\" onclick=\"deselect_dacscans()\">DESELECT ALL</button>" << std::endl;
+    *out << "    </div>"<< std::endl;
+    *out << "</div>" << std::endl; // end panel
+}
+
+
+
+void gem::calib::CalibrationWeb::settingsInterface(calType_t m_calType, xgi::Output* out, xdata::Integer m_nShelves)
+    throw (xgi::exception::Exception)
+{
+    *out << "<div id=\"cal_interface\">" << std::endl;
+    *out << "    <div align=\"center\">" << std::endl;
+    *out << "        <div class=\"alert alert-warning\">" << std::endl;
+    *out << "            <button type=\"button\" class=\"close\" data-dismiss=\"alert\">&times;</button>" << alertMap.find(m_calType)->second << std::endl;
+    *out << "        </div>" << std::endl;
+    *out << "    </div>" << std::endl;
+    *out << "    <div class=\"row\">" << std::endl;
+    *out << "        <div class=\"col-md-6\">" << std::endl;
+    *out << "            <form id=\"settings_select\">" << std::endl;
+
+    // create a temporary parameters map,
+    // check whether requested calibration type needs a trigger selector
+    // if it is, create it and pop out the trigger setting
+    std::map<std::string, std::string> t_parameters = dynamic_cast<gem::calib::Calibration*>(p_gemApp)->m_scanParams.find(m_calType)->second; 
+
+    for (auto parameter: t_parameters) {
+        if (dynamic_cast<gem::calib::Calibration*>(p_gemApp)->m_scanParamsNonForm.find(parameter.first) !=dynamic_cast<gem::calib::Calibration*>(p_gemApp)->m_scanParamsNonForm.end()) {
```

Comment:
As you use this conversion multiple times in this function (and in other places where you are doing multiple casts), probably better to create a local variable, e.g.,
```c++
auto gemAppLocal = dynamic_cast<gem::calib::Calibration*>(p_gemApp);
```